modules/porous_flow/src/auxkernels/PorousFlowElementLength.C

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowElementLength.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowElementLength);
+
+InputParameters
+PorousFlowElementLength::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+
+  params.addCoupledVar("direction",
+                       "Direction (3-component vector) along which to compute the length");
+
+  params.addClassDescription(
+      "AuxKernel to compute the 'length' of elements along a given direction.  A plane is "
+      "constructed through the element's centroid, with normal equal to the direction given.  The "
+      "average of the distance of the nodal positions to this plane is the 'length' returned.  The "
+      "Variable for this AuxKernel must be an elemental Variable");
+
+  return params;
+}
+
+PorousFlowElementLength::PorousFlowElementLength(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _num_direction(coupledComponents("direction")),
+    _direction_x(coupledValue("direction", 0)),
+    _direction_y(coupledValue("direction", std::min(_num_direction - 1, (unsigned)1))),
```

Comment:
Isn't `_num_direction` always 3? So `_num_direction - 1` is always 2? Or have I missed something (quite possible!)

Reply:
Couldn't a user just shove in one coupled component?   Of course they should specify 3, but they might erroneously shove in less.

Reply:
But you error out below (line 40) if it isn't 3?

Reply:
Won't MOOSE just crash if `num_direction=1` and i use
```
_direction_y(coupledValue("direction", 1))
```

Reply:
So you are trying to avoid just a crash if the user mistakenly enters one or two values in `direction`, but then throw an informative error later on? I think I get it now. Unless that is wrong, in which case I don't get it still!

Reply:
Yea, that's correct.  Sorry for not explaining it better

Reply:
Out of interest, do you know what happens if you make `direction` a `RealVectorValue` param instead like we do for gravity in assorted kernels? Does it error out with a useful message if you don't specify three components? I don't recall ever trying this.

Reply:
Yes, i originally wrote it this way, BUT i need separate coupledVariables because i use MultiApps to transfer them, hence the current version.

Reply:
Makes sense now.